ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -166,8 +166,12 @@ public void setViewState(@NonNull T view, @Nullable ReadableMap accessibilitySta
     if (accessibilityState == null) {
       return;
     }
+    if (accessibilityState.hasKey("selected")) {
+      view.setSelected(accessibilityState.getBoolean("selected"));
```

Comment:
the java test case does not verify that `view.setSelected(true)` is invoked, but just verifies `view.setTag(R.id.accessibility_state, accessibilityState)`

Reply:
https://github.com/facebook/react-native/commit/099be9b35634851b178e990c47358c2129c0dd7d

Reply:
The following test:

```java
  @Test
  public void testAccessibilityStateSelected() {
    WritableMap accessibilityState = Arguments.createMap();
    mView = mock(ReactViewGroup.class);
    accessibilityState.putBoolean("selected", true);
    mViewManager.setViewState(mView, accessibilityState);
    assertThat(mView.getTag(R.id.accessibility_state)).isEqualTo(accessibilityState);
    verify(mView, Mockito.times(1)).setSelected(true);
  }
```
triggers the error:

```
FAILURE com.facebook.react.uimanager.BaseViewManagerTest testAccessibilityStateSelected: loader constraint violation: loader net.bytebuddy.dynamic.loading.MultipleParentClassLoader @688614c0 wants to load interface com.android.internal.util.Predicate. A different interface with the same name was previously loaded by org.robolectric.internal.AndroidSandbox$SdkSandboxClassLoader @23f7d05d. (com.android.internal.util.Predicate is in unnamed module of loader org.robolectric.internal.AndroidSandbox$SdkSandboxClassLoader @23f7d05d, parent loader java.net.FactoryURLClassLoader @728938a9)
```

seems that I can not mock `mView` so set the expectation

https://github.com/facebook/react-native/blob/b02a4266d89bd85f58cbb884d895baec24aa78c2/ReactAndroid/src/test/java/com/facebook/react/uimanager/BaseViewManagerTest.java#L46

the sourcecode of the [method](https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/android/view/View.java) is:

```java
    /**
     * Changes the selection state of this view. A view can be selected or not.
     * Note that selection is not the same as focus. Views are typically
     * selected in the context of an AdapterView like ListView or GridView;
     * the selected view is the view that is highlighted.
     *
     * @param selected true if the view must be selected, false otherwise
     */
    public void setSelected(boolean selected) {
        //noinspection DoubleNegation
        if (((mPrivateFlags & PFLAG_SELECTED) != 0) != selected) {
            mPrivateFlags = (mPrivateFlags & ~PFLAG_SELECTED) | (selected ? PFLAG_SELECTED : 0);
            if (!selected) resetPressedState();
            invalidate(true);
            refreshDrawableState();
            dispatchSetSelected(selected);
            if (selected) {
                sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_SELECTED);
            } else {
                notifyViewAccessibilityStateChangedIfNeeded(
                        AccessibilityEvent.CONTENT_CHANGE_TYPE_UNDEFINED);
            }
        }
    }
```

Reply:
Thanks a lot! Solved with commit https://github.com/facebook/react-native/pull/31144/commits/79486b50cb7821e4af0796f43232688e96fc4d7a

The test discussed would not fail in the master branch as it does not verify the new behaviour added with pr #31144 in java. 
The test previously would just verify that the `mView` had a tag `R.id.accessibility_state` with value equal to variable `accessibilityState`, but this behaviour was not changed in this pr.

```java
assertThat(mView.getTag(R.id.accessibility_state)).isEqualTo(accessibilityState)
````

https://github.com/facebook/react-native/blob/a6cc5871e68b63e2f29bcb3610968dd33b926694/ReactAndroid/src/test/java/com/facebook/react/uimanager/BaseViewManagerTest.java#L72-L77

Now the test verifies that `mView.isSelected()` is true after passing `accessibilityState.selected` true.
I previously tried to run this verification by mocking `mView` instance, but it is not possible and I later found the public method in the Android sourcecode.

https://developer.android.com/reference/android/view/View#isSelected()
```java
    /**
     * Indicates the selection state of this view.
     *
     * @return true if the view is selected, false otherwise
     */
    @ViewDebug.ExportedProperty
    @InspectableProperty(hasAttributeId = false)
    public boolean isSelected() {
        return (mPrivateFlags & PFLAG_SELECTED) != 0;
    }
```

Related to https://github.com/facebook/react-native/pull/31144/files#r595064322